### ISSUE TITLE
Missing characters on first line when powershell starts

### ISSUE THREAD

**Issue by Balkoth** [JanDeDobbeleer/oh-my-posh#864]
![image](https://user-images.githubusercontent.com/80055480/126643567-850c8d87-46ce-4b6e-95eb-dcb99b5eb9ed.png)

![image](https://user-images.githubusercontent.com/80055480/126643591-d89c23bb-5dd0-4aaa-bfeb-a7d68b0a1e8b.png)

### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] I have looked for duplicate issues before submitting this one

### Description

The first line of a new windows terminal powershell instance misses out on some characters.

### Environment

- Oh My Posh version: 3.169.1
- Theme: e.g.: powerline, powerlevel10k_rainbow
- Operating System: Windows 10 Version 2004 (Build 19041.1083)
- Terminal: Windows Terminal Preview  1.10.1933.0
- PowerShell: 5.1.19041 / 7.1.3

#### Optional

- posh-git version: 1.0.0
- git version: 2.32.0.windows.2

**Comment by JanDeDobbeleer** [comment]
@Balkoth that's a Windows Terminal bug. The fix can be found [here](https://ohmyposh.dev/docs/faq#windows-terminal-unexpected-space-between-segmentstext).

**Comment by Balkoth** [comment]
Sorry but the fix does not work for me. Version 2 does not have this problem.

**Comment by JanDeDobbeleer** [comment]
@Balkoth the bug is not in oh-my-posh, but Windows Terminal and icons/ANSI. So if it doesn't work, please follow up on the linked [issue](https://github.com/microsoft/terminal/issues/3546) so they can fix it.

**Comment by david-baierl** [comment]
For my workes adding `echo ''` at the end of my `.bashrc` file to force a new line bevor printing anything

**Comment by hyoretsu** [comment]
> @Balkoth that's a Windows Terminal bug. The fix can be found [here](https://ohmyposh.dev/docs/faq#windows-terminal-unexpected-space-between-segmentstext).

The worked for me. In my case, whitespace at the end wasn't rendering the background color. At first Windows Terminal displayed it correctly, but after some time it didn't.

**Comment by t0m5k1** [comment]
I too face this issue but I'm lost as to how I fix it...

> @Balkoth that's a Windows Terminal bug. The fix can be found [here](https://ohmyposh.dev/docs/faq#windows-terminal-unexpected-space-between-segmentstext).

This is great but where do I put the code from this FAQ ?
settings.json or profile.ps1 or some other file I'm not familiar with??

I feel this needs more scope on what people should actually be doing with the code snippet rather than blindly giving it out and expecting people to know exactly where it belongs

**Comment by JanDeDobbeleer** [comment]
@t0m5k1 this needs to be added to the configuration. In case you're using a theme, export it first using [this](https://ohmyposh.dev/docs/installation/customize#adjust-a-theme) guide. I do see an issue with the snippet in the FAQ, `template` is now part of the root object rather than `properties`:

```json
{
  "type": "executiontime",
  "template": "\ufbab\u2800" // invisible spacing character
}
```

**Comment by t0m5k1** [comment]
Think I'll just live with the gap.

**Comment by JanDeDobbeleer** [comment]
@t0m5k1 why? It's not that hard, follow the steps and adjust the template accordingly (or ask the WT team to fix their terminal 😁)

**Comment by t0m5k1** [comment]
Because I've done what you said and added the 4 lines of code but it's not taking effect. Seeing as you created this I guess I have wider issues that I just don't have the time to look into.
I've no idea what I'm doing, I'm not a coder in any way and this at then end of the day is eye candy which can be lived with as PS is still usable.

**Comment by JanDeDobbeleer** [comment]
@t0m5k1 can you share the config you adjusted? I'll help you.

**Comment by t0m5k1** [comment]
Here you go:
https://pastebin.com/B61xTesV

**Comment by JanDeDobbeleer** [comment]
@t0m5k1 can you show me a screenshot of where the gap is? That way I can adjust the config accordingly. From what I can see in the shared config, is that you seem to be confused as to where to adjust the correct `template`. You added the [example](https://github.com/JanDeDobbeleer/oh-my-posh/issues/864#issuecomment-1519631174) I gave at random places, which won't do what you're trying to achieve. I'll try to explain it better, but it's also advised to [read the docs](https://ohmyposh.dev/docs/configuration/overview) on how this all works, otherwise you'll get confused even more when trying to adjust other elements.

Every segment has a template that defines how the content is rendered. In your example, the first block has the `session` segment first, defined like:

```json
{
  "type": "session",
  "style": "diamond",
  "foreground": "p:black",
  "background": "p:yellow",
  "leading_diamond": "\ue0b6",
  "trailing_diamond": "\ue0b0",
  "template": " {{ if .SSHSession }}\uf817 {{ end }}{{ .UserName }} "
}
```

Here, the template says it will print the `\uf817` icon if we have an active `.SSHSession`, see each [segment template](https://ohmyposh.dev/docs/segments/session#properties) properties for possibilities here. Suppose that the `\uf817` icon has a space after it, like what you see in your prompt, we can use the invisible character hack to have Windows Terminal display it correctly by appending `\u2800` to the broken character:

```json
{
  "type": "session",
  "style": "diamond",
  "foreground": "p:black",
  "background": "p:yellow",
  "leading_diamond": "\ue0b6",
  "trailing_diamond": "\ue0b0",
  "template": " {{ if .SSHSession }}\uf817\u2800 {{ end }}{{ .UserName }} "
}
```

So, in your case, check where the prompt has that space, which is always after a specific character and add it to that template, after the broken character and it should be solved. If not, change the icon to a different one, there's a whole [cheat sheet](https://www.nerdfonts.com/cheat-sheet) for this over at Nerd Fonts.

**Comment by t0m5k1** [comment]
Hi Jan,
Thanks for pointing out where I went wrong with this. I think it would be better if the FAQ gave some additional detail in so much as that it should state that every single segment needs to have this added because that's what lead me to do what I did!
I also stated "I'm not a coder in any way!" Therefore I wouldn't have the slightest idea where to put the template.

As I said previously I'll just live with it the way it is, If I go poking around any further I'll have even less of a clue of what I did ..I'm kinda getting used to it and it is just eye candy. Additionally I have a feeling I'll need to reverse all this when MS decide to fix what ever is broken.
Cheers
